### PR TITLE
[JENKINS-49584] - Plugin facelift

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] Update Plugin POM and cleanup dependencies
- [x] Update core requirement to 1.625.3. This version is almost 4 years old, should be fine
- [x] Add Jenkinsfile

@ljader let me know if you are not interested in maintaining this plugin anymore. In my team we maintain several plugins depending on JSch (e.g. Maven Plugin), so we could take it.

@reviewbybees @ljader

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -74,46 +71,16 @@
...
             <artifactId>ssh-credentials</artifactId>
-            <version>${deps.ssh-credentials-plugin.version}</version>
+            <version>1.13</version>
         </dependency>
 
-        <!-- test dependencies -->
-        <dependency>
-            <groupId>org.apache.sshd</groupId>
-            <artifactId>sshd-core</artifactId>
-            <version>${deps.apache-sshd.version}</version>
```
Comment (oleg-nenashev) [inline pom.xml]: 1.625.3 bundles SSHD Core which is new enough, so we do not need it anymore

**Comment by oleg-nenashev** [comment]
@reviewbybees done

File: pom.xml [hunk 2]
```diff
@@ -3,15 +3,15 @@
...
+        <version>3.4</version>
     </parent>
 
     <artifactId>jsch</artifactId>
     <version>0.1.54.2-SNAPSHOT</version>
     <packaging>hpi</packaging>
 
     <name>Jenkins JSch dependency plugin</name>
-    <url>http://wiki.jenkins-ci.org/display/JENKINS/JSch+plugin</url>
+    <url>http://wiki.jenkins.io/display/JENKINS/JSch+plugin</url>
```
Comment (dwnusbaum) [inline pom.xml]: Should this be `https`?

**Comment by ljader** [comment]
@oleg-nenashev please go ahead and take over maintaining of the plugin.
I'm not up to date with recent Jenkins plugin cleanup work (whitelisting etc).

**Comment by oleg-nenashev** [comment]
@daniel-beck created https://issues.jenkins-ci.org/browse/JENKINS-49584 for this warning

**Pull request merged**